kani-compiler/src/kani_middle/transform/body.rs

```diff
@@ -224,6 +225,15 @@ impl MutableBody {
             }
         }
     }
+
+    /// Clear all the existing logic of this body and turn it into a simple `return`.
+    ///
+    /// We do not prune the local variables today for simplicity.
```

Comment:
not sure I understand the question

Reply:
@feliperodri? Do you mean why we don't prune them? I can change that if you prefer.

Reply:
Yes. I actually want to apply the same logic to all our intrinsics. Today, we have 2 implementations of many of our intrinsics, one for concrete execution and one for verification. This is to avoid bringing extra logic to the verification, such as this change: https://github.com/model-checking/kani/pull/2032.

If we can clear the body of our intrinsic functions for verification, like we do here, we no longer care about what is inside the function, and they can have real logic.

Reply:
I'd would include this detail on the documentation, because it's a very clear "**why** we have this function", while the documentation of the function today only says "**what** this function does". Thanks for the clarification.

Reply:
Please see the new suggestion below.